data/fin.lean

```diff
@@ -88,3 +91,12 @@ by cases i; refl
   {n} {C : fin (succ n) → Sort*} {H0 Hs} (i : fin n) :
   @fin.cases n C H0 Hs i.succ = Hs i :=
 by cases i; refl
+
+def raise_nat {n} (i : fin n) (k) : fin (n + k) :=
+⟨i.val, lt_add_of_lt_of_nonneg i.is_lt (nat.zero_le k)⟩
+
+def lower_left {n m} (i : fin (n + m)) (h : i.val < n) : fin n :=
+⟨i.val, by apply h⟩
```

Comment:
This proof can also be shorter. Because of @spl's request, I'll just give a hint: the proof I'm thinking of has one character.

Reply:
Done.